rts/Lua/LuaMatrix.hpp

```diff
@@ -0,0 +1,197 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+#ifndef LUA_MATRIX_H
+#define LUA_MATRIX_H
+
+#include <map>
+#include <vector>
+#include <string>
+#include <tuple>
+
+#include "System/Log/ILog.h"
+
+#include "lib/sol2/sol.hpp"
+#include "lib/fmt/format.h"
+
+#include "System/Matrix44f.h"
+#include "System/MathConstants.h"
+#include "System/EventClient.h"
+#include "System/float4.h"
+
+#include "Sim/Features/Feature.h"
+#include "Sim/Units/Unit.h"
+
+#undef near
+#undef far
+
+struct lua_State;
+
+
+struct CSolidObject;
```

Comment:
This type of forward declaration be unnecessary as you're already indirectly including it through Sim/Units/Unit.h 

Additionally, as CSolidObject is defined as a class, we should also forward declare it as a class, and not a struct